clippy_lints/src/semicolon_if_nothing_returned.rs

```diff
@@ -52,14 +52,19 @@ impl<'tcx> LateLintPass<'tcx> for SemicolonIfNothingReturned {
                 if let ExprKind::DropTemps(..) = &expr.kind {
                     return;
                 }
-                span_lint_and_sugg(
+                span_lint_and_then(
```

Comment:
This is actually unnecessary, you only need to change `app` to `MachineApplicable`.

Reply:
oh, it is magic :)
thanks
pushed!